client/trino-cli/src/main/java/io/trino/cli/StatusPrinter.java

```diff
@@ -324,10 +324,13 @@ private void printQueryInfo(QueryStatusInfo results, WarningsPrinter warningsPri
             int progressWidth = (min(terminalWidth, 100) - 75) + 17; // progress bar is 17-42 characters wide
 
             if (stats.isScheduled()) {
-                String progressBar = formatProgressBar(progressWidth,
-                        stats.getCompletedSplits(),
-                        max(0, stats.getRunningSplits()),
-                        stats.getTotalSplits());
+                int completed = stats.getCompletedSplits();
+                int running = stats.getRunningSplits();
+                int total = stats.getTotalSplits();
+                if (progressPercentage > 0) {
```

Comment:
This is to prevent from progress rolling back, right?

Reply:
yes